paymentsheet/src/main/java/com/stripe/android/paymentsheet/IntentConfirmationHandler.kt

```diff
@@ -492,18 +453,25 @@ internal class IntentConfirmationHandler(
     }
 
     private fun onPaymentResult(result: InternalPaymentResult) {
-        val intentResult = when (result) {
-            is InternalPaymentResult.Completed -> PaymentConfirmationResult.Succeeded(
-                intent = result.intent,
-                deferredIntentConfirmationType = deferredIntentConfirmationType
+        val intentResult = runCatching {
+            val arguments = currentArguments ?: throw IllegalStateException(
+                "Arguments should have been initialized before handling payment result!"
             )
-            is InternalPaymentResult.Failed -> PaymentConfirmationResult.Failed(
-                cause = result.throwable,
-                message = result.throwable.stripeErrorMessage(),
-                type = PaymentConfirmationErrorType.Payment,
+
+            val intentConfirmationOption = arguments.confirmationOption as? PaymentConfirmationOption.PaymentMethod
+                ?: throw IllegalStateException("Cannot confirm intent with non payment method confirmation option")
```

Comment:
is there a way we can avoid throwing these IllegalStateExceptions? e.g. can we include the `confirmationOption` in the `IntentPaymentResult` and just get it from there?

Reply:
I talked with Jay regarding this, we definitely want to reach a state where we do this. It just requires work to update all our launcher types to return the confirmation option as part of the result or at least return the necessary data to produce the confirmation option. I plan on doing this work following adding the registry.